src/redist/targets/GenerateLayout.targets

```diff
@@ -335,7 +335,7 @@
 
     <DownloadFile Condition=" '@(ComponentToDownload)' != '' And '%(ComponentToDownload.ShouldDownload)' == 'true'"
                   Uri="%(ComponentToDownload.BaseUrl)/%(ComponentToDownload.DownloadFileName)"
-                  PrivateUri="%(ComponentToDownload.PrivateBaseUrl)/%(ComponentToDownload.DownloadFileName)$(DOTNETCLIMSRC_READ_SAS_TOKEN)"
+                  PrivateUri="%(ComponentToDownload.PrivateBaseUrl)/%(ComponentToDownload.DownloadFileName)$([System.Environment]::GetEnvironmentVariable('DOTNETCLIMSRC_READ_SAS_TOKEN'))"
```

Comment:
It seems like the only cause for the error could be this line, right @joeloff? I don't see how the version upgrades could cause the sourcebuild issue.

Reply:
Yup, I agree. There were some minor devops issues, so I kicked the build again. I seem to recall there's a way that custom tasks can fail, not return false, but still register the failure so MSBUILD thinks there was an error.

Reply:
@mmitche it looks like you added this change 4 months ago, does it need to flow into main?

Reply:
Yeah it probably should, otherwise we have do it again in .NET 7

Reply:
Do you know if it could be causing the errors we're seeing in CI?

Reply:
I suppose it's possible. But the PrivateUri is only used if Uri is unable to be accessed. It does look weird though..the binlog has a failed build but no errors. Also there appears to only be one usage of DownloadFile and it's not even passing that parameter.

Btw, I believe installer should switch to this and remove their own implementation
https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.Arcade.Sdk/src/DownloadFile.cs